File: frontend/types/notification.ts
export type NotificationType =
    | "new_application"
    | "new_message"
    | "application_status_changed"
    | "invitation"
    | "new_vacancy";


export interface Notification {
    id: number;

    recipient_id: number;

    sender_id: number | null;

    sender_name: string | null;

    type: NotificationType;

    title: string;

    message: string;

    is_read: boolean;

    application_id: number | null;

    conversation_id: number | null;

    vacancy_id: number | null;

    created_at: string;

    updated_at: string;
}

File: backend/app/schemas/notification.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from backend.app.models.enums import NotificationType


class NotificationResponse(BaseModel):
    id: int

    recipient_id: int

    sender_id: int | None
    sender_name: str | None

    type: NotificationType

    title: str

    message: str

    is_read: bool

    application_id: int | None

    conversation_id: int | None

    vacancy_id: int | None

    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(
        from_attributes=True,
    )


class NotificationReadResponse(BaseModel):
    success: bool